pkg/limitador/k8s_objects.go

```diff
@@ -180,20 +180,28 @@ func DeploymentName(limitadorObj *limitadorv1alpha1.Limitador) string {
 }
 
 func PodDisruptionBudget(limitadorObj *limitadorv1alpha1.Limitador) *policyv1.PodDisruptionBudget {
-	return &policyv1.PodDisruptionBudget{
+	pdb := &policyv1.PodDisruptionBudget{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      PodDisruptionBudgetName(limitadorObj),
 			Namespace: limitadorObj.ObjectMeta.Namespace,
 			Labels:    Labels(limitadorObj),
 		},
 		Spec: policyv1.PodDisruptionBudgetSpec{
-			MaxUnavailable: limitadorObj.Spec.PodDisruptionBudget.MaxUnavailable,
-			MinAvailable:   limitadorObj.Spec.PodDisruptionBudget.MinAvailable,
 			Selector: &metav1.LabelSelector{
 				MatchLabels: Labels(limitadorObj),
 			},
 		},
 	}
+
+	if limitadorObj.Spec.PodDisruptionBudget == nil {
+		helpers.TagObjectToDelete(pdb)
+		return pdb
+	}
```

Comment:
I have a question that raise a concern with me on this. If I understand PR will mean that we always create the PDB CR even if it is not being used. When it is not being used we mark it to be delete. Is it possible for us to get into an infinite loop here?

 If we create a resource we should be reconciling any changes the user might make to that resource. By this logic if the resource is delete from the cluster we should check to see if that is the correct state, and with this change we would recreate the resource marking it for deletion causing the loop to start over. If we are not watching the resource for change causing use to not be affected by the infinite loop, then the question becomes how do we guarantee the resource is in the condition stated in the limitador CR?

Reply:
when we "tag" object to be deleted, the reconciler will make sure it is deleted and if it exists, it will delete it. https://github.com/Kuadrant/kuadrant-operator/blob/main/pkg/reconcilers/base_reconciler.go#L110-L127

Objects tagged as deleted will never be created